### PR TITLE
Updated prestashop/ps_emailsubscription to 2.5.0, prestashop/ps_facetedsearch to 3.2.1, prestashop/ps_contactinfo to 3.2.0

### LINKED ISSUE #9907: [BOOM-3749] Problème d'affichage sur module natif greffé
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-3749).

- _**Reporter:**_ bernon
- _**Created at:**_ Thu, 31 Aug 2017 18:09:16 +0200

<p>Création d'une colonne gauche sur la page index.<br/>
Module Newsletter greffé sur la colonne de Gauche et remonté en première position.</p>
        
* How to reproduce the issue ?

N/A

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Updated prestashop/ps_emailsubscription to 2.5.0<br />Updated prestashop/ps_facetedsearch to 3.2.1<br />Updated prestashop/ps_contactinfo to 3.2.0 
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #9907 
| How to test?  | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15354)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]
Thanx @Progi1984 !

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
composer.json needs to be updated too.

**Comment by Progi1984** [comment]
@PierreRambaud Why ? Composer.json is correct :
```
        "prestashop/ps_contactinfo": "^3",
        "prestashop/ps_emailsubscription": "^2",
        "prestashop/ps_facetedsearch": "^3",
```

**Comment by PierreRambaud** [comment]
Not really, PrestaShop's not working if we use the current version of our modules :)

**Comment by Progi1984** [comment]
@PierreRambaud It's why we update the composer.lock. Composer install will base on this file.

**Comment by PierreRambaud** [comment]
@Progi1984 But the first thing you check when you want to know which version you're using is the composer.json.
If the app isn't working anymore with an old version of a module, we must also update the composer.json. Otherwise it's like having ">0" everywhere and only trust the composer.lock.

**Comment by Progi1984** [comment]
@PierreRambaud @matthieu-rolland It should be better :)

File: composer.json [hunk 1]
```diff
@@ -52,13 +52,13 @@
...
-        "prestashop/ps_contactinfo": "^3",
+        "prestashop/ps_contactinfo": "^3.2",
         "prestashop/ps_currencyselector": "^2",
         "prestashop/ps_customeraccountlinks": "^3",
         "prestashop/ps_customersignin": "^2",
         "prestashop/ps_customtext": "^4",
-        "prestashop/ps_emailsubscription": "^2",
-        "prestashop/ps_facetedsearch": "^3",
+        "prestashop/ps_emailsubscription": "^2.5",
+        "prestashop/ps_facetedsearch": "^3.2",
```
Comment (PierreRambaud) [inline composer.json]:
```suggestion
        "prestashop/ps_facetedsearch": "^3.2.1",
```
The 3.2.0 introduces many bugs :)

Comment (Progi1984) [reply composer.json]: :ok_hand:

**Comment by Progi1984** [comment]
@PierreRambaud Corrected

**Review by PierreRambaud — APPROVED** [review]

**Comment by pierreloicq** [comment]
Hi,
I am unsure how it relates to this PR, but at the moment, the upgrade proposed for ps_emailsubscription on 1.7.6.1 lead to something buggy on classic theme: e-mail is not saved in the db, and there is no "well received" message.
Thanks

**Comment by Progi1984** [comment]
@pierreloicq Which version of ps_emailsubscription do you have ?

**Comment by pierreloicq** [comment]
@Progi1984  2.3 provided with PS 1.7.61. was working fine, then I upgraded to 2.5 which is buggy

**Comment by sarahdib** [comment]
Hello @pierreloicq 

Thank you for your feed back. 

I just test (twice) the module email_subscription and it's working for me the email is saved and I receive the email for confirmation.

In the module configuration did you enable the option "send a confirmation email" ?

Have a nice day :)

**Comment by pierreloicq** [comment]
@sarahdib No I did not enable it(I was talking about the message in green that appears below the e-mail box after you submit). But if I activated it, I receive nothing. 
Actually I have an error popping up when I go on the module config page : 
```
Notice on line 74 in file C:\\Program Files (x86)\\Ampps\\www\\presta17\\src\\Adapter\\Routing\\LegacyHelperLinkBuilder.php
[8] Undefined index: id_merged
```

**Comment by Progi1984** [comment]
@pierreloicq Thanks for your feedback. Could you create a specific issue for this ? Thanks.

**Comment by pierreloicq** [comment]
Here it is
https://github.com/PrestaShop/PrestaShop/issues/15425

**Comment by matks** [comment]
Thank you @Progi1984

**Pull request merged**